### PR TITLE
Rewrite shader in WGSL

### PR DESCRIPTION
**Pull request by kylc**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -226,32 +226,43 @@ impl Texture {
...
 /// Configuration for the renderer.
-pub struct RendererConfig<'vs, 'fs> {
+pub struct RendererConfig<'s> {
     pub texture_format: TextureFormat,
     pub depth_format: Option<TextureFormat>,
     pub sample_count: u32,
-    pub vertex_shader: Option<ShaderModuleDescriptor<'vs>>,
-    pub fragment_shader: Option<ShaderModuleDescriptor<'fs>>,
+    pub shader: Option<ShaderModuleDescriptor<'s>>,
+    pub color_space: ColorSpace,
```
Comment (cwfitzgerald) [inline src/lib.rs:241]: We should ask for a vertex_shader_entry_point and fragment_shader_entry_point that way we don't force the idea of caring about spaces onto the user.

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]
Shaders look good, one comment about the surrounding code.

File: src/lib.rs (hunk 1 above)
Comment (kylc) [reply src/lib.rs]: You're right, that seems more general. See the latest commit.

File: src/lib.rs [hunk 2]
```diff
@@ -227,31 +231,30 @@ impl Texture {
...
 /// Configuration for the renderer.
-pub struct RendererConfig<'vs, 'fs> {
+pub struct RendererConfig<'s> {
     pub texture_format: TextureFormat,
     pub depth_format: Option<TextureFormat>,
     pub sample_count: u32,
-    pub vertex_shader: Option<ShaderModuleDescriptor<'vs>>,
-    pub fragment_shader: Option<ShaderModuleDescriptor<'fs>>,
+    pub shader: Option<ShaderModuleDescriptor<'s>>,
+    pub vertex_shader_entry_point: String,
```
Comment (cwfitzgerald) [inline src/lib.rs:239]: This should just be a &str, we don't expect this struct to live a long time so adding the lifetimes are okay.

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]
One last thing then we should be good.

File: src/lib.rs (hunk 2 above)
Comment (kylc) [reply src/lib.rs]: The `RenderConfig` is currently copied into the `Renderer` and must live as long. The shader uses an interesting pattern of swapping for `None` and assigning a `'static` lifetime, so I mirrored that pattern for the shader entry points. It does mean the user can pass `None` for the entry points which will crash (as they always could with the shader), but it's at least consistent with the existing API.

Comment (cwfitzgerald) [reply src/lib.rs]: Alright, I think that's weird, but totally shouldn't block this.

**Review by cwfitzgerald — APPROVED** [review]

**Pull request merged**